routers/api/v1/packages/package.go

```diff
@@ -213,3 +218,96 @@ func ListPackageFiles(ctx *context.APIContext) {
 
 	ctx.JSON(http.StatusOK, apiPackageFiles)
 }
+
+// LinkPackage sets a repository link for a package
+func LinkPackage(ctx *context.APIContext) {
+	// swagger:operation POST /packages/{owner}/{type}/{name}/link package linkPackage
+	// ---
+	// summary: Link a package to a repository
+	// parameters:
+	// - name: owner
+	//   in: path
+	//   description: owner of the package
+	//   type: string
+	//   required: true
+	// - name: type
+	//   in: path
+	//   description: type of the package
+	//   type: string
+	//   required: true
+	// - name: name
+	//   in: path
+	//   description: name of the package
+	//   type: string
+	//   required: true
+	// - name: repo_id
+	//   in: query
+	//   description: ID of the repository to link. Pass `0` to unlink
+	//   type: integer
+	//   required: false
+	// - name: repo_name
+	//   in: query
+	//   description: name of the repository to link, format `{owner}/{name}`
+	//   type: string
+	//   required: false
+	// responses:
+	//   "201":
+	//     "$ref": "#/responses/empty"
+	//   "404":
+	//     "$ref": "#/responses/notFound"
+
+	pkg, err := packages.GetPackageByName(ctx, ctx.ContextUser.ID, packages.Type(ctx.Params("type")), ctx.Params("name"))
+	if err != nil {
+		if errors.Is(err, util.ErrNotExist) {
+			ctx.Error(http.StatusNotFound, "GetPackageByName", err)
+		} else {
+			ctx.Error(http.StatusInternalServerError, "GetPackageByName", err)
+		}
+		return
+	}
+
+	var repoID int64
+
+	formRepoID := ctx.FormString("repo_id")
```

Comment:
Do we really want to expose the repository's id? I don't think we should do that. Using `OwnerName` and `RepoName` should be better.

Reply:
I'm not sure what you mean by "exposing" in this context. The existing `GET /repos/{owner}/{repo}` endpoint makes the `id` value available in its response, so that information is already "exposed" to every consumer of the public API.

What's the harm in giving the user a choice which identifier they can run this new endpoint against?

Reply:
@sclu1034 I think last comments refer to the fact that most, if not all the API, refer to orgs and repos using their name, and not their id. At least I already played with the API and I don't recall I ever sent requests using the repo ids. It just feels more natural to use the repo name instead.

Moreover, the web frontend only allow linking to a repository of the same owner. If that is by design, you would have to drop the owner part in the repo name.

Just my last two cents on the endpoint: there could only be zero or one link to a repository, so I'd rather use PUT and DELETE.

Reply:
> Moreover, the web frontend only allow linking to a repository of the same owner. If that is by design, you would have to drop the owner part in the repo name.
I think this is by design. It's weird to link a package to another owner's repository.

Reply:
@lunny well I'm still wondering. I develop some Unity (game engine) packages. Distribution relies on NPM protocol, although packages are obviously not meant for Nodejs (it's C#). And I also develop some Nodejs packages. But Gitea only provides a single NPM registry per owner. So it seems to me the only way to cleanly separate both usages is to make a dedicated "owner", so to have two registries. But this is only for distribution. Development still occurs in the same org.